packages/app/src/runs/RunsContainer.vue

```diff
@@ -92,30 +94,95 @@ fragment RunsContainer on Query {
   ...RunsConnect
 }`
 
-const props = defineProps<{
-  gql: RunsContainerFragment
-}>()
+gql`
+mutation RunsContainer_FetchNewerRuns(
+  $cloudProjectNodeId: ID!, 
+  $beforeCursor: String, 
+  $hasBeforeCursor: Boolean!,
+  $refreshPendingRuns: [ID!]!,
+  $hasRefreshPendingRuns: Boolean!
+) {
+  refetchRemote {
+    cloudNode(id: $cloudProjectNodeId) {
+      id
+      __typename
+      ... on CloudProject {
+        runs(first: 10) @skip(if: $hasBeforeCursor) {
+          ...RunsContainer_RunsConnection
+        }
+        newerRuns: runs(last: 10, before: $beforeCursor) @include(if: $hasBeforeCursor) {
+          ...RunsContainer_RunsConnection
+        }
+      }
+    }
+    cloudNodesByIds(ids: $refreshPendingRuns) @include(if: $hasRefreshPendingRuns) {
+      id
+      ... on CloudRun {
+        ...RunCard
+      }
+    }
+  }
+}
+`
 
-const isCloudProjectReturned = computed(() => props.gql.currentProject?.cloudProject?.__typename === 'CloudProject')
+const currentProject = computed(() => props.gql.currentProject)
+const cloudViewer = computed(() => props.gql.cloudViewer)
 
-const isOnlineRef = ref(true)
-const showConnectSuccessAlert = ref(false)
+const variables = computed(() => {
+  if (currentProject.value?.cloudProject?.__typename === 'CloudProject') {
+    const toRefresh = currentProject.value?.cloudProject.runs?.nodes?.map((r) => r.status === 'RUNNING' ? r.id : null).filter((f) => f) ?? []
 
-watchEffect(() => {
-  // We want to keep track of the previous state to refetch the query
-  // when the internet connection is back
-  if (!online.value && isOnlineRef.value) {
-    isOnlineRef.value = false
+    return {
+      cloudProjectNodeId: currentProject.value?.cloudProject.id,
+      beforeCursor: currentProject.value?.cloudProject.runs?.pageInfo.startCursor,
+      hasBeforeCursor: Boolean(currentProject.value?.cloudProject.runs?.pageInfo.startCursor),
+      refreshPendingRuns: toRefresh,
+      hasRefreshPendingRuns: toRefresh.length > 0,
+    }
   }
 
-  if (online.value && !isOnlineRef.value) {
-    isOnlineRef.value = true
-    emit('reexecuteRunsQuery')
+  return undefined as any
+})
+
+const refetcher = useMutation(RunsContainer_FetchNewerRunsDocument)
+
+// 15 seconds polling
+const POLL_FOR_LATEST = 1000 * 15
+const timeout = ref<null | number>(null)
```

Comment:
No need for `ref`, can just use `let` - this doesn't seem to need to be reactive.